packages/cheerio-crawler/src/internals/cheerio-crawler.ts

```diff
@@ -1062,23 +1065,39 @@ export class CheerioCrawler<JSONData = unknown> extends BasicCrawler<
 interface EnqueueLinksInternalOptions {
     options?: CheerioCrawlerEnqueueLinksOptions;
     $: CheerioRoot | null;
-    requestQueue?: RequestQueue;
-    defaultBaseUrl?: string;
+    requestQueue: RequestQueue;
+    originalRequestUrlOrigin?: string;
+    finalRequestUrlOrigin?: string;
 }
 
 /** @internal */
-export async function cheerioCrawlerEnqueueLinks({ options, $, requestQueue, defaultBaseUrl }: EnqueueLinksInternalOptions) {
+export async function cheerioCrawlerEnqueueLinks({ options, $, requestQueue, originalRequestUrlOrigin, finalRequestUrlOrigin }: EnqueueLinksInternalOptions) {
     if (!$) {
         throw new Error('Cannot enqueue links because the DOM is not available.');
     }
 
-    const baseUrl = options?.baseUrl ?? defaultBaseUrl;
+    let baseUrl: string | undefined;
+
+    if (options?.baseUrl) {
+        // User provided base url takes priority
+        baseUrl = options.baseUrl;
+    } else if (options?.strategy === EnqueueStrategy.All) {
+        // We can assume users want to go off the domain in this case
+        baseUrl = finalRequestUrlOrigin ?? originalRequestUrlOrigin;
+    } else if (originalRequestUrlOrigin === finalRequestUrlOrigin) {
+        // If the domains are the same, we can use the original domain
+        baseUrl = originalRequestUrlOrigin;
+    }
+    // else {
+    // TODO: Do we want to use finalRequestUrlOrigin if domains match or leave it as undefined?
```

Comment:
not sure about this, can you give me an example on some edge cases where this matters?

Reply:
The only time I could think this would be an edge case is `domain.com/redirect` -> `subdomain.domain.com` but I'm not too sure... That would not have a baseUrl set so no links would be enqueued

Reply:
ok, will leave it up to you, i guess we dont need to handle all edge cases right now, especially when we are not sure they exist. we will catch issues like this when we start testing on platform and in real projects.

Reply:
Should we leave the TODO for now so we don't forget or clean it up and handle it when issues actually arise?

Reply:
i would either keep it uncommented or remove it fully (together with the unused dependency, unless its used elsewhere too) 

maybe better to have the baseUrl inferred all the time, i am sure we will catch it early if it would be problematic, so i'd be in favour of uncommenting